jdk/test/sun/security/ssl/X509TrustManagerImpl/distrust/Distrust.java

```diff
@@ -25,169 +25,142 @@
-        "entrust2048ca", "affirmtrustcommercialca", "affirmtrustnetworkingca",
-        "affirmtrustpremiumca", "affirmtrustpremiumeccca" };
+    private final boolean before;
+    private final boolean policyOn;
+    private final boolean isValid;
 
-    // A date that is after the restrictions take effect
-    private static final Date NOVEMBER_12_2024 =
-        Date.from(LocalDate.of(2024, 11, 12)
-                           .atStartOfDay(ZoneOffset.UTC)
-                           .toInstant());
-
-    // A date that is a second before the restrictions take effect
-    private static final Date BEFORE_NOVEMBER_12_2024 =
-        Date.from(LocalDate.of(2024, 11, 12)
-                           .atStartOfDay(ZoneOffset.UTC)
-                           .minusSeconds(1)
-                           .toInstant());
-
-    public static void main(String[] args) throws Exception {
-
-        cf = CertificateFactory.getInstance("X.509");
-
-        boolean before = args[0].equals("before");
-        boolean policyOn = args[1].equals("policyOn");
-        boolean isValid = args[2].equals("valid");
+    public Distrust(String[] args) {
+        before = args[0].equals("before");
+        policyOn = args[1].equals("policyOn");
+        isValid = args[2].equals("valid");
 
         if (!policyOn) {
             // disable policy (default is on)
             Security.setProperty("jdk.security.caDistrustPolicies", "");
         }
+    }
 
-        Date notBefore = before ? BEFORE_NOVEMBER_12_2024 : NOVEMBER_12_2024;
+    public Date getNotBefore(ZonedDateTime distrustDate) {
+        ZonedDateTime notBefore = before ? distrustDate.minusSeconds(1) : distrustDate;
+        return Date.from(notBefore.toInstant());
+    }
 
-        X509TrustManager pkixTM = getTMF("PKIX", null);
-        X509TrustManager sunX509TM = getTMF("SunX509", null);
-        for (String test : rootsToTest) {
-            System.err.println("Testing " + test);
-            X509Certificate[] chain = loadCertificateChain(test);
+    public void testCodeSigningChain(String certPath, String name, Date validationDate)
+            throws Exception {
+        System.err.println("Testing " + name + " code-signing chain");
+        Validator v = Validator.getInstance(Validator.TYPE_PKIX,
+                Validator.VAR_CODE_SIGNING,
+                getParams());
+        // set validation date so this will still pass when cert expires
+        v.setValidationDate(validationDate);
+        v.validate(loadCertificateChain(certPath, name));
+    }
 
-            testTM(sunX509TM, chain, notBefore, isValid);
-            testTM(pkixTM, chain, notBefore, isValid);
+    public void testCertificateChain(String certPath, Date notBefore, X509TrustManager[] tms,
+                                     String... tests) throws Exception {
+        for (String test : tests) {
+            System.err.println("Testing " + test);
+            X509Certificate[] chain = loadCertificateChain(certPath, test);
```

Comment:
Nit: if we want to maximize the similarity with the `jdk11u-dev` version, we should add a blank line after this one (`Distrust.java:78`).

See `Distrust.java` diff against `jdk11u-dev` (before openjdk/jdk11u-dev@8322c66efa9da9210eca7d6081d2a8c2d65ba4e0) in the previous comment.